packages/bigtable/src/table.js

```diff
@@ -669,16 +687,27 @@ Table.prototype.insert = function(entries, callback) {
  *     deleted.
  * @param {function} callback - The callback function.
  * @param {?error} callback.err - An error returned while making this request.
- * @param {object} callback.apiResponse - The full API response.
+ * @param {object[]} callback.statuses - A status for each entity transaction.
  *
  * @example
  * //-
  * // Insert entities. See {module:bigtable/table#insert}
  * //-
- * var callback = function(err, apiResponse) {
- *   if (!err) {
- *     // Mutations were successful.
+ * var callback = function(err, statuses) {
+ *   if (err) {
+ *     // Error handling omitted.
  *   }
+ *
+ *   // statuses = [
```

Comment:
That should still work with the model of telling the user what failed instead of what succeeded.

Streaming mode:
```js
table.mutate(entries)
  .on('error', function(err) {
    err = API error OR {
      message: 'Could not insert entry at index 50',
      entry: { the original entry }
    }
  })
  .on('finish', function() {
    // All entries processed by API
    // Entries that couldn't be saved were emitted to the `error` event handler
  })
```

Callback version (some users won't care about the streaming and just want the results all at once):

```js
table.mutate(entries, function(err, insertErrors) {
  if (err) {
    // API returned an error
  }

  if (insertErrors.length > 0) {
    // These entries were not saved
    insertErrors = [
      {
        message: 'Could not insert entry at index 50'
        entry: { the original entry }
      }
    ]
  }
})
```